### PR TITLE
feat: new socket code

### PR DESCRIPTION
**Pull request by abretonc7s**
## Explanation

Refactor socket server code to:
- Enable redis inside socket io code (it was using memory adapter instead of redis adapter so could never work)
- Enable authentication for socket in sdk-communication-layer and prioritize websocket connection over long polling.
- Refactor ts errors and linter config
- Add logger

Create new project `packages/devsocket` to allow interaction with socket server and validate scalability.

## References

None.

### `@metamask/sdk-communication-layer`

- **BREAKING**: new socket server initialization with sticky session

## Checklist

- [x] I've updated the test suite for new or updated code as appropriate
- [x] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [x] I've highlighted breaking changes using the "BREAKING" category above as appropriate

### CODE REVIEW

**Comment by abretonc7s** [comment]
@SocketSecurity ignore-al

**Comment by abretonc7s** [comment]
https://github.com/SocketSecurity ignore-all

**Comment by abretonc7s** [comment]
@SocketSecurity ignore-all

File: packages/sdk-socket-server-next/src/index.ts [hunk 1]
```diff
@@ -5,29 +5,51 @@ import dotenv from 'dotenv';
...
+      res.set('Content-Type', 'text/plain');
+      const totalClients = ioServer.engine.clientsCount;
+      let fullMetrics = `# HELP socket_io_server_total_clients Total number of connected clients\n`;
+      fullMetrics += `# TYPE socket_io_server_total_clients gauge\n`;
+      fullMetrics += `socket_io_server_total_clients ${totalClients}\n`;
+
+      const totalRooms = ioServer.sockets.adapter.rooms.size;
+      fullMetrics += `# HELP socket_io_server_total_rooms Total number of rooms\n`;
+      fullMetrics += `# TYPE socket_io_server_total_rooms gauge\n`;
+      fullMetrics += `socket_io_server_total_rooms ${totalRooms}\n`;
```
Comment (omridan159) [inline packages/sdk-socket-server-next/src/index.ts:43]:
```suggestion
      const metrics = buildMetrics(ioServer);
```

WDYT about extracting this logic into a function?

**Review by omridan159 — APPROVED** [review]

**Pull request merged**